From 3c5025548707edabdf334961aa47027d590fd5a7 Mon Sep 17 00:00:00 2001
From: Daniel Ammann <daniel.ammann@bytesatwork.ch>
Date: Thu, 14 Jul 2022 10:14:39 +0200
Subject: [PATCH] dt-bindings: arm: bytesatwork: Add module and board

Document the bytesatwork AG vendor prefix.
Add byteENGINE and byteDEVKIT.

Signed-off-by: Daniel Ammann <daniel.ammann@bytesatwork.ch>
---
 .../bindings/arm/bytesatwork/bytesatwork.yaml | 30 +++++++++++++++++++
 .../devicetree/bindings/vendor-prefixes.yaml  |  2 ++
 MAINTAINERS                                   |  1 +
 3 files changed, 33 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml

diff --git a/Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml b/Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
new file mode 100644
index 000000000000..2587e61c70e4
--- /dev/null
+++ b/Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
@@ -0,0 +1,30 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/arm/bytesatwork/bytesatwork.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: bytesatwork Platforms Device Tree Bindings
+
+maintainers:
+  - Daniel Ammann <daniel.ammann@bytesatwork.ch>
+
+properties:
+  $nodename:
+    const: "/"
+  compatible:
+    oneOf:
+      - description: bytesatwork byteENGINE
+        items:
+          - enum:
+              - bytesatwork,byteengine-am62x
+          - const: bytesatwork,byteengine-am62x
+      - description: bytesatwork byteDEVKIT
+        items:
+          - enum:
+              - bytesatwork,bytedevkit-am62x
+          - const: bytesatwork,bytedevkit-am62x
+
+additionalProperties: true
+
+...
diff --git a/Documentation/devicetree/bindings/vendor-prefixes.yaml b/Documentation/devicetree/bindings/vendor-prefixes.yaml
index f1d1882009ba..0db2ab15c0bd 100644
--- a/Documentation/devicetree/bindings/vendor-prefixes.yaml
+++ b/Documentation/devicetree/bindings/vendor-prefixes.yaml
@@ -261,6 +261,8 @@ patternProperties:
     description: B&R Industrial Automation GmbH
   "^bytedance,.*":
     description: ByteDance Ltd.
+  "^bytesatwork,.*":
+    description: bytesatwork AG
   "^calamp,.*":
     description: CalAmp Corp.
   "^calao,.*":
diff --git a/MAINTAINERS b/MAINTAINERS
index 9a9a3dd3951b..b1500f62d8ff 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -5385,6 +5385,7 @@ M:	Daniel Ammann <daniel.ammann@bytesatwork.ch>
 S:	Maintained
 F:	.github/workflows/checkpatch.yml
 F:	.gitlab-ci.yml
+F:	Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
 
 C-MEDIA CMI8788 DRIVER
 M:	Clemens Ladisch <clemens@ladisch.de>
